### DISCUSSION TITLE
Photoshop Format shows images distorted

### DISCUSSION THREAD

**Discussion by styler2go** [ImageMagick/ImageMagick#3608] [Help] [answered]
Hello, 

when i try to export an image into psd format i have the problem that the resulting image looks distorted while the text looks fine:
![image](https://user-images.githubusercontent.com/2152144/116376984-0a8f3580-a811-11eb-91ad-dc8d95c7b5d9.png)

Any ideas what could cause this?

The command is as follows:
```
convert  -size "300x200" \
\( xc:none -font "Nimbus-Mono-L-Bold" -gravity "Center" -pointsize "16" -fill "#00ff00" -annotate  45x45+0+0 "Tempora placeat." \) \
\( xc:none -font "Nimbus-Mono-L-Bold" -gravity "Center" -pointsize "16" -fill "#ff0000" -annotate  45x45+40+0 "385,00 €" \) \
\( xc:none -gravity "Center" -virtual-pixel "transparent" -draw "image over 0,0,200 200 'image-2-test2.jpg'" -background "none" -quality "100" -mosaic -distort ScaleRotateTranslate "45" -set colorspace RGB \) \
\( xc:#FF0000  \) \
\(  -clone 0-4 -flatten \) \
-reverse psd:- > photo.psd
```

**Comment by snibgo** [comment]
What version of IM, on what platform?

What is the simplest command that shows the problem?

Please show your input image image-2-test2.jpg.

**Reply by styler2go** [reply]
- I tried multiple Imagemagick Versions: 7.0.10.43 for windows did NOT have this problem, Other version above this (even on windows) had this issue.
- image-2-test2.jpg can be any image, i tried multiple images.
- Simpelst command could be: `convert  -size "200x200" \( xc:none -virtual-pixel "transparent" -draw "image over 0,0,200 200 'image-2-test2.jpg'" \) \(  -clone 0-4 -flatten \) -reverse psd:- > photo.psd`


![image-2-test2](https://user-images.githubusercontent.com/2152144/116416310-8651a800-a83a-11eb-8ba7-99352a313ce2.jpg)

**Comment by fmw42** [comment]
This command makes no sense. You only have one input image, yet you are cloning 5 images to flatten.

**Reply by styler2go** [reply]
Are you responding to the short simplified command above? If so, then yes, i missed to fix that when copy-pasting. It won't fix the issue though.

**Comment by snibgo** [comment]
Try a command that is more simplified:
```
magick -size 200x200 xc:None -virtual-pixel "transparent" -draw "image over 0,0,200 200 image-2-test2.jpg" x.png
```
Does that show the problem? If so, that eliminates issues with layers, PSD format, etc.

**Reply by styler2go** [reply]
This problem only occurs for psd format. PNGs are not affected and will render out correctly. Your command therefore works for png and psd. The problem only starts to occure when trying to generate multiple layer PSDs

**Comment by fmw42** [comment, marked as answer]
I did some tests and it looks like PS does not like having a composite within the command. It complains about a non-supported blending method. -compose over should be the same as blending of normal, but when accepting normal, it shows your issue.

If I remove the composite and correct some of your commands to remove extraneous things, then this seems to work OK:

magick -size "300x200" \
\( xc:none -font "Nimbus-Mono-L-Bold" -gravity "Center" -pointsize "16" -fill "#00ff00" -annotate  45x45+0+0 "Tempora placeat." \) \
\( xc:none -font "Nimbus-Mono-L-Bold" -gravity "Center" -pointsize "16" -fill "#ff0000" -annotate  45x45+40+0 "385,00 €" \) \
\( 'image-2-test2.jpg' -virtual-pixel "transparent" -distort SRT "45" \) \
\( xc:#FF0000  \) \
\(  -clone 0-3 -background none -flatten \) \
-reverse x.psd

A PNG export of the image, looks as follows:

![psd1](https://user-images.githubusercontent.com/13458218/116441267-293bff00-a806-11eb-80a6-a2c15599119f.png)


However, another issue occurs if I try to resize the image in the command. It seems to produce a doubled image when imported into PS.

magick -size "300x200" \
\( xc:none -font "Nimbus-Mono-L-Bold" -gravity "Center" -pointsize "16" -fill "#00ff00" -annotate  45x45+0+0 "Tempora placeat." \) \
\( xc:none -font "Nimbus-Mono-L-Bold" -gravity "Center" -pointsize "16" -fill "#ff0000" -annotate  45x45+40+0 "385,00 €" \) \
\( 'image-2-test2.jpg' -virtual-pixel "transparent" -distort SRT "45" -resize 300x200 \) \
\( xc:#FF0000  \) \
\(  -clone 0-3 -background none -flatten \) \
-reverse x.psd

![psd2](https://user-images.githubusercontent.com/13458218/116441405-4a9ceb00-a806-11eb-91f5-9a492fe3798c.png)

**Reply by styler2go** [reply]
Using just `( 'image-2-test2.jpg' -virtual-pixel "transparent" -distort SRT "45" -resize 300x200 )`, how can i set the position of the image?

**Reply by fmw42** [reply]
Set the position relative to what?  That is not a full proper command. Are you doing anything else in the command besides reading the input and writing the output. Please always give a proper full command.

**Reply by fmw42** [reply]
See See https://legacy.imagemagick.org/Usage/distorts/#distort_viewport

**Reply by styler2go** [reply]
> Set the position relative to what? That is not a full proper command. Are you doing anything else in the command besides reading the input and writing the output. Please always give a proper full command.

I ment setting the position relative to it's gravity and size of the image. For example, in my draw-command the image was centered but 20px below the top of the whole image:
` -draw "image over 20,0,200 200 'image-2-test2.jpg'"`

In general, i just want to position the image on the surface of the whole image using a position (x,y) and gravity

**Reply by fmw42** [reply]
Use -geometry +X+Y to set the offset relative to gravity.

**Reply by styler2go** [reply]
![whitestar](https://user-images.githubusercontent.com/2152144/117096625-f8118080-ad69-11eb-808d-b8b863497eb9.png)
It doesn't seem to work with just using -geometry. See this command:
`convert -size "1330x259" xc:"transparent" ( "C:\server\htdocs\bct-banner-creation-tool\backend\public\whitestar.png" -gravity "NorthWest" -geometry "+300+0" ) -flatten rating.png`

**Reply by fmw42** [reply]
Gravity and geometry need to be outside the parentheses

**Comment by fmw42** [comment]
The IM developers should look into this. I can work with them on it, if they want.

**Reply by styler2go** [reply]
I am experiencing exactly the same issue regarding duplicated images. Can we somehow report this to the IM developers?
